libraries/NDP/src/NDP.cpp

```diff
@@ -239,24 +213,39 @@ int NDPClass::sendData(uint8_t *data, unsigned int len)
   return s;
 }
 
-
+int NDPClass::extractStart(void) {
+  uint32_t sample_size;
```

Comment:
make `sample_size`a member of the class (can probably be set default to 768 in init), then we can set the member variable in `extractStart()`